frontend/packages/pipelines-plugin/src/components/catalog/apis/artifactHub.ts

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { CatalogItem } from '@console/dynamic-plugin-sdk';
+import { K8sResourceKind, k8sCreate, k8sUpdate } from '@console/internal/module/k8s';
+import { consoleProxyFetchJSON } from '@console/shared/src/utils/proxy';
+import { ARTIFACTHUB_API_BASE_URL } from '../../../const';
+import { TaskModel, TaskModelV1Beta1 } from '../../../models';
+import { TektonTaskAnnotation } from '../../pipelines/const';
+import { ARTIFACTHUB } from '../../quicksearch/const';
+import { getInstalledFromAnnotation } from '../../quicksearch/pipeline-quicksearch-utils';
+import { ApiResult } from '../hooks/useApiResponse';
+
+export type ArtifactHubRepository = {
+  name: string;
+  kind: number;
+  url: string;
+  display_name: string;
+  repository_id: string;
+  organization_name: string;
+  organization_display_name: string;
+};
+
+export type ArtifactHubVersion = {
+  version: string;
+  contains_security_update: boolean;
+  prerelease: boolean;
+  ts: number;
+};
+
+export type ArtifactHubTask = {
+  package_id: string;
+  name: string;
+  description: string;
+  version: string;
+  display_name: string;
+  repository: ArtifactHubRepository;
+};
+
+export type ArtifactHubTaskDetails = {
+  package_id: string;
+  name: string;
+  description: string;
+  display_name: string;
+  keywords: string[];
+  platforms: string[];
+  version: ArtifactHubVersion[];
+  available_version: [];
+  content_url: string;
+  repository: ArtifactHubRepository;
+};
+
+const ARTIFACRHUB_TASKS_SEARCH_URL = `${ARTIFACTHUB_API_BASE_URL}/packages/search?offset=0&limit=60&facets=false&kind=7&deprecated=false&sort=relevance`;
+
+export const getArtifactHubTaskDetails = async (item: CatalogItem): Promise<any> => {
+  const API_BASE_URL = `${ARTIFACTHUB_API_BASE_URL}/packages/tekton-task`;
+  const { name, data } = item;
+  const {
+    task: {
+      version,
+      repository: { name: repoName },
+    },
+  } = data;
+  const url = `${API_BASE_URL}/${repoName}/${name}/${version}`;
+  return consoleProxyFetchJSON({ url, method: 'GET' });
+};
+
+export const useGetArtifactHubTasks = (hasPermission: boolean): ApiResult<any> => {
+  const [resultData, setResult] = React.useState([]);
```

Comment:
Will this work? Untested

```suggestion
export const useGetArtifactHubTasks = (hasPermission: boolean): ApiResult<ArtifactHubTask[]> => {
  const [resultData, setResult] = React.useState<ArtifactHubTask[]>([]);
```

Reply:
Yes, it will work